drivers/nvram_spi/nvram-spi.c

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2015 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+#include "nvram.h"
+#include "nvram-spi.h"
+#include "byteorder.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "hwtimer.h"
+
+/**
+ * @ingroup     nvram
+ * @{
+ *
+ * @file
+ *
+ * @brief       Device interface for various SPI connected NVRAM.
+ *
+ * Tested on:
+ * - Cypress/Ramtron FM25L04B.
+ *
+ * @author      Joakim Gebart <joakim.gebart@eistec.se>
+ */
+
+typedef enum {
+    /** WRITE command byte, 0b0000 0010 */
+    NVRAM_SPI_CMD_WRITE = 0x02,
+    /** READ command byte, 0b0000 0011 */
+    NVRAM_SPI_CMD_READ = 0x03,
+    /** WREN command byte, 0b0000 0110 */
+    NVRAM_SPI_CMD_WREN = 0x06,
+} nvram_spi_commands_t;
```

Comment:
I think they are the same opcodes as used in EEPROMs. It seems like these opcodes are common for many manufacturers. At least Fujitsu FRAM, Everspin MRAM, Cypress FRAM, Micron NVSRAM, Adesto CBRAM, Rohm FRAM, are using them.